cache/cache_helpers.py

```diff
@@ -86,6 +96,29 @@ def _increment_stat(self, stat: Statistic, char: str):
                 stat.watcher_count += 1
         stat.all_character_count += 1
 
+    def clear(self):
+        self.__init__()
+
+class RunStatsByDate(RunStats):
+    def __init__(self):
+        super().__init__()
+        self.start_date = None
+        self.end_date = None
+
+    @property
+    def date_range_string(self) -> str:
+        date_str = ""
+        format = "%Y/%m/%d"
+        if self.start_date is None and self.end_date is None:
+            date_str = "all-time"
```

Comment:
not technically "all-time", but it is for as long as the profile's earliest run in late 2020.